ui/src/app/app.tsx

```diff
@@ -97,6 +101,20 @@ export class App extends React.Component<{}, { popupProps: PopupProps }> {
     }
 
     public render() {
+        if (this.state.error != null) {
+            const stack = this.state.error.stack;
+            const url = 'https://github.com/argoproj/argo-cd/issues/new?labels=bug&template=bug_report.md';
+
+            return (
+                <React.Fragment>
```

Comment:
Minor, did we want to use https://reactjs.org/docs/error-boundaries.html?

Reply:
Duh. I can see you did.

Reply:
The docs say that if a class defines either or both of `static getDerivedStateFromError()` or `componentDidCatch()`, then it becomes an error boundary. I used `getDerivedStateFromError()` here.